apps/guardian-ui/src/components/GuardianAuthenticationCode.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import {
+  Text,
+  useTheme,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Flex,
+  Button,
+} from '@chakra-ui/react';
+import { useTranslation } from '@fedimint/utils';
+import QRCode from 'qrcode.react';
+
+const QR_CODE_SIZE = 256;
+
+interface GuardianAuthenticationCodeProps {
+  federationId: string;
+  ourPeer: { id: number; name: string };
+}
+
+export const GuardianAuthenticationCode: React.FC<
+  GuardianAuthenticationCodeProps
+> = ({ federationId, ourPeer }) => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [qrValue, setQrValue] = useState<string>('');
+  const [isAcknowledged, setIsAcknowledged] = useState(false);
+
+  const calculateGuardianAuthenticationCode = () => {
+    const password = sessionStorage.getItem('guardian-ui-key');
```

Comment:
Maybe try-catch this to be safe? just in case the storage key doesn't exist for some reason?